### ISSUE TITLE
LibreELEC 9.0.1 Shadertoy (1.1.9.1) Visualisation, preset list empty

### ISSUE THREAD

**Issue by x2k13** [xbmc/visualization.shadertoy#35]
Hello @notspiff ,

I've recently installed LibreELEC 9.0.1 on my Raspberry Pi 3 B+ and looked forward to have those nice shadertoy visualisations again, like they used to be.

Just one issue: the preset list is empty and the only visualisation that seems to get triggered/works is that full screen level indicator. The preset files are indeed located in the correct folder of the installed addon.

I tested the screensaver addon, that one has presets available in the list and they seem to work all fine, running smoothly on this nifty arm device. Both addons were installed from the LibreELEC repo. Any ideas on this?

All the best!

**Comment by Rechi** [comment]
Please post your complete kodi log file with debug logging enabled (https://kodi.wiki/view/Log_file).

In which directory are the preset located?

**Comment by x2k13** [comment]
The presets are located in ~/.kodi/addons/visualization.shadertoy/resources/

ls -la here: 
https://vim.cx/?343dc1a5b7931356#UP6siaXrEYdTsM7lFJuHJC1DplGExJl1rJD7rFEiZCs=

IIRC the visualisations shoul change on track change, at least. That's not happening at all.

Here's the debug-log: 
https://vim.cx/?58e46cabe83ac22e#q+QxbdbyVN01+Ggpyn8UCMGN5k/BrHbjH9+zUNU+xp4=

I hope you guys can find what's the culprit. Oh and I forgot to add that the adon's settings just have one entry (button), Programs. If you click that, nothing happens.

All the best

**Comment by x2k13** [comment]
@notspiff @Rechi any progress yet?

**Comment by weiserhei** [comment]
Im running Kodi 18.3 on RPI4 and had the same issue, only the level indicator was avaliable. 

**Installing the Shadertoy screensaver like you mentioned in your first post fixed it for me.** (Settings -> Interface -> Screensaver -> Screensaver mode -> Get more... -> Shadertoy).
Apparently after that I could run the visualisations on music again. `Choose presets` is populated now with all the well known vis's. 

`Visualisation Settings` which leads to `Programs` is still empty, so there is nothing so customize.

![screenshot001](https://user-images.githubusercontent.com/7581732/60773999-a405c880-a10d-11e9-846d-367ac3e6d8db.png)

**Comment by x2k13** [comment]
@weiserhei thanks for poining this out! 

In the background of your screenshot I could see that you used the settings-cogwheel of the player controlbar to get to the presets. That does the trick for me too!

The only thing missing is a "random" setting in there and some of the presets don't work on my Pi 3B+ ... but that's another story.

Cheers!

**Comment by Cris--** [comment]
> 
> 
> Im running Kodi 18.3 on RPI4 and had the same issue, only the level indicator was avaliable.
> 
> **Installing the Shadertoy screensaver like you mentioned in your first post fixed it for me.** (Settings -> Interface -> Screensaver -> Screensaver mode -> Get more... -> Shadertoy).
> Apparently after that I could run the visualisations on music again. `Choose presets` is populated now with all the well known vis's.
> 
> `Visualisation Settings` which leads to `Programs` is still empty, so there is nothing so customize.


Doesn't found a way to PM @weiserhei so here my question:
Have you running a working shadertoy version on **RPi4** / LibreElec / Kodi?

**Comment by weiserhei** [comment]
Yes, shadertoy is running on RPI4, also tested right now with the new 9.2 Beta 2.

**Comment by Cris--** [comment]
Can you tell me please how do you get this working? I'm also on LE 9.2 Beta 2 with the ShaderToy 2.0.0.l4 from the LE Repro. IF i start the screensaver i get a black screen only.

Better would be the music visualization ShaderToy addon.  ^^  (..or projectM)

**Comment by Cris--** [comment]
I'm also on LE 9.2b2 and with this method over "Get more..." installation i get a blackscreen only on screensaver start. If i try to start the music visualization ShaderToy i get a "Unknown Error.." popup.

Screensaver ShaderToy v2.0.0.4 LE repo
Music Vis ShaderToy v1.0.8 popcronmix repo

some [log informations (pastebin)...](https://pastebin.com/8mJYYVeH)

**EDIT:** Strange, after a reboot and some idle time (ShaderToy was set as screensaver) it runs now.
Kewl! But some new presets (slower, not so intensive. More like a Lava Lamp) would be great.

**Comment by stuaxo** [comment]
Had the same symptom on Android TV - (no presets until I installed via the screensaver option), not sure if it's changing eithe.

I don't have time to poke @ the logs, but hopefully will next time I see this :)
